selendroid-server/src/main/java/io/selendroid/server/action/ActionChain.java

```diff
@@ -18,10 +18,11 @@
   private int pauseTime = 0;
   private int index = 0;
 
-  public ActionChain(JSONObject actionChain) throws JSONException {
+  public ActionChain(JSONObject actionChain, int i) throws JSONException {
     this.inputDevice = actionChain.getString("inputDevice");
     this.actionChain = actionChain.getJSONArray("actions");
     this.context = new ActionContext();
+    this.context.setId(i);
```

Comment:
seems like these id's could get accidentally re-used with subsequent calls and could have some potential adverse affects?

Why don't we use a UUID and set it here?
http://developer.android.com/reference/java/util/UUID.html#randomUUID()

Then the method signature can go back to just taking a JSONObject